src/redirect/server.js

```diff
@@ -0,0 +1,158 @@
+client to URL where file/Sharing Manifest is exposed.
+*/
+app.post('/qr/:id/claim', (req, res) => {
+  const qrId = req.params.id;
+  logger.info(`Sending POST request to /qr/${qrId}/claim`);
+  // Get QR details using given QR id
+  const policy = QRs.get(qrId);
+
+  if (!policy || policy.claims.length >= (policy.request.claimsLimit || Infinity)) {
+    res.status(403);
+    return res.json(`QR ${qrId} is not valid or has already been claimed`);
+  }
+
+  const clientSpecificUrl = `/qr/${qrId}/claimed/${randomId()}`;
+
+  // create new claim
+  policy.claims.push({
+    clientName: req.query.clientName || 'unknown',
+    clientSpecificUrl,
+    queryLog: [`Claimed: ${new Date()}`],
+    locationAlias: Object.fromEntries(
+      Array.from(new Set(policy.request.access.flatMap(a => a.locations || [])).values()).map(l => [l, randomId()])
+    )
+  });
+
+  // redirect to 301 Moved Permanently if redemption is successful
+  res.redirect(301, clientSpecificUrl);
+});
+
+/*
+GET to clientSpecificURL for SHCLink that was claimed.
+
+Allows client to access location where SHCLink file/Sharing Manifest is exposed.
+*/
+app.get('/qr/:id/claimed/:cid', (req, res) => {
+  const { cid, id } = req.params;
+  logger.info(`Sending GET request to /qr/${id}/claimed/${cid}`);
+  const policy = QRs.get(id);
+
+  if (!policy) {
+    res.status(403);
+    return res.json(`QR ${id} is no longer valid`);
+  }
+
+  const claimDetails = policy.claims.find(c => c.clientSpecificUrl === req.url);
+  if (!claimDetails) {
+    res.status(403);
+    return res.json(`This QR has not been correctly claimed`);
+  }
+
+  res.json(
+    policy.request.access.map(a => ({
+      ...a,
+      locations: a.locations?.map(l => `${PUBLIC_URL}/qr/${id}/claimed/${cid}/files/${claimDetails.locationAlias[l]}`)
+    }))
+  );
+});
+
+/*
+GET claimed file.
+*/
+app.get('/qr/:id/claimed/:cid/files/:fileid', async (req, res) => {
+  const { id, cid, fileid } = req.params;
+  logger.info(`Sending GET request to /qr/${id}/claimed/${cid}/files/${fileid}`);
+  const policy = QRs.get(id);
+
+  if (!policy) {
+    res.status(403);
+    return res.json(`QR ${id} is no longer valid`);
+  }
+
+  const claimDetails = policy.claims.find(c => c.clientSpecificUrl === `/qr/${id}/claimed/${cid}`);
+  if (!claimDetails) {
+    res.status(403);
+    return res.json(`This QR has not been correctly claimed`);
+  }
+
+  const trueLocation = Object.entries(claimDetails.locationAlias).find(
+    ([, clientSpecific]) => clientSpecific === fileid
+  )[0];
+
+  logger.debug(`trueLocation found for id ${id}`);
+
+  const proxied = await fetch(trueLocation);
+  res.status(proxied.status);
+  res.header('Content-Type', proxied.headers.get('content-type') || 'application/text');
+  res.send(await proxied.text());
+});
+
+/*
+Used for testing functionality of above endpoints. Serves as fake
+static file hosting for testing.
+*/
+app.get('/:file.json', async (req, res) => {
```

Comment:
The browser gets hung up if you make a request to a file that doesn't exist (try navigating to `http://localhost:3000/example3.json`).

Consider adding an else statement top the end of this to process responses to requests for files that don't exist within the `test/redirect` folder.

For Example:
```
  } else {
    logger.debug(`file path ${filePath} does not exist`);
    res.status(404).send(`File does not exist`);
  }
```